internal/manifests/collector/configmap.go

```diff
@@ -18,16 +18,21 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/open-telemetry/opentelemetry-operator/internal/api/convert"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/manifestutils"
 	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
 )
 
 func ConfigMap(params manifests.Params) (*corev1.ConfigMap, error) {
-	name := naming.ConfigMap(params.OtelCol.Name)
-	labels := manifestutils.Labels(params.OtelCol.ObjectMeta, name, params.OtelCol.Spec.Image, ComponentOpenTelemetryCollector, []string{})
+	otelCol, err := convert.V1Alpha1to2(params.OtelCol)
```

Comment:
Instead of converting in each method, wouldn't it simpler to create the `manifest.Params` structure using the `v1alpha2` version of the collector?

Reply:
Sure, thats the final goal. But to do this know, we need to adapt ta, opamp and other packages too.

- https://github.com/open-telemetry/opentelemetry-operator/pull/2532#discussion_r1458043087
- https://github.com/open-telemetry/opentelemetry-operator/pull/2532#issue-2087265240